asf_heat_pump_suitability/pipeline/run_scripts/run_add_features.py

```diff
@@ -85,10 +85,11 @@ def parse_arguments() -> argparse.Namespace:
     logging.info("Adding lat/lon data to EPC")
     uprn_latlon_df = lat_lon.transform_df_osopen_uprn_latlon()
     epc_df = epc_df.join(uprn_latlon_df, how="left", on="UPRN")
+    epc_gdf = lat_lon.generate_gdf_uprn_coords(epc_df, usecols=["UPRN", "lad_code"])
 
     # Replace `lad_code` from postcode with `lad_code` from geospatial join and postcode
     logging.info("Adding LAD code with geospatial join")
-    uprn_lad_df = output_areas.sjoin_df_uprn_lad_code(epc_df)
+    uprn_lad_df = output_areas.sjoin_df_uprn_lad_code(epc_gdf)
```

Comment:
I think (?) it has to be an `sjoin_nearest` because with `sjoin`, these are the predicates for joining:
https://geopandas.org/en/stable/docs/user_guide/mergingdata.html#binary-predicate-joins
- intersects
- contains
- within
- touches
- crosses
- overlaps
So points that aren't exactly touching/ on top of each other will not get joined together. This was the case with the Wales listed buildings sjoin that was in the code before. It does join some of the buildings, but you get much more with the `sjoin_nearest` methodology. Does this make sense?

Reply:
yeh makes sense! Thanks for a explanation :)